schema-mapping/feature-types/appearance/AbstractTexture.json

```diff
@@ -0,0 +1,57 @@
+{
+  "identifier": "app:AbstractTexture",
+  "table": "surface_data",
+  "properties": [
+    {
+      "name": "imageURI",
+      "namespace": "http://3dcitydb.org/3dcitydb/appearance/5.0",
+      "value": {
+        "column": "image_uri",
+        "type": "uri"
+      },
+      "join": {
+        "table": "tex_image",
+        "fromColumn": "tex_image_id",
+        "toColumn": "id"
+      }
+    },
+    {
+      "name": "mimeType",
+      "namespace": "http://3dcitydb.org/3dcitydb/appearance/5.0",
+      "value": {
+        "column": "mime_type",
+        "type": "string"
+      },
+      "join": {
+        "table": "tex_image",
+        "fromColumn": "tex_image_id",
+        "toColumn": "id"
+      }
+    },
+    {
+      "name": "codeSpace",
```

Comment:
is here also a `join` to `tex_image` needed for codespace?

Reply:
Well, `codeSpace` is a child property of `mimeType`. For accessing `mimeType`, a join to `tex_image` is requried. Thus, it is implicitly assumed that `codeSpace` is also stored in `tex_image`. So, a `join` switches the context to the new table, and all further properties are assumed to be stored in this new table. You would only have to put a join for `codeSpace` if it was stored in another table than `tex_image`.